### PR TITLE
Add build test CI using github actions

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
**Problem Description**
This PR adds a build test CI for this repository using github actions.

This adds a nice sanity checks if the chain of catkin dependencies are being broken and if we need to start fixing dependency versions.

While adding the PR the following was added

- Simulation dependencies seems to be broken and ill configured for rosdep installations. Since they are also not part of the dependency list, I think it is more sensible to remove them by default from the `package.xml` and add them as optional. I have separated the commits to show that the dependencies are broken for the simulation
- `grpc` is a dependency of `protobuf_catkin` but was not part of the dependency list and was not getting pulled
- The build test runs both ROS Melodic(Ubuntu Bionic 18.04) and ROS Noetic(Ubuntu Focal 20.04), but Noetic fails due to voxblox not supporting ROS Noetic. This is due to pcl requiring `c++14` and higher and is being addressed in https://github.com/ethz-asl/voxblox/pull/379

Since the build is not automatically triggered in the `ethz-asl` organization from forks, you can see the build results in https://github.com/Jaeyoung-Lim/mav_active_3d_planning/pull/1

image

**Additional Context**
- The build is configured to only be triggered by new merges and pull requests

### CODE REVIEW

**Review by Schmluk — APPROVED** [review]
As dicussed

**Comment by Jaeyoung-Lim** [comment]
@Schmluk Thanks!

**Comment by Jaeyoung-Lim** [comment]
@Schmluk Voxblox and `protobuf_catkin` now supports ROS Noetic and therefore re-enabled Noetic build for this repo

**Comment by Schmluk** [comment]
Nice! LGTM, feel free to merge!

**Comment by Jaeyoung-Lim** [comment]
@Schmluk I don't have merge rights! :sweat_smile:

**Comment by Schmluk** [comment]
Interesting I thought you should be allowed to once the PR is approved. I merged the PR and added you to the repo as a maintainer for the future :)

**Comment by Jaeyoung-Lim** [comment]
@Schmluk Thanks!

**Pull request merged**
